sdk/java/pulumi/src/test/java/com/pulumi/serialization/internal/JumboCustomType.java

```diff
@@ -0,0 +1,3111 @@
+    	      this.bar34 = defaults.bar34;
+    	      this.bar35 = defaults.bar35;
+    	      this.bar36 = defaults.bar36;
+    	      this.bar37 = defaults.bar37;
+    	      this.bar38 = defaults.bar38;
+    	      this.bar39 = defaults.bar39;
+    	      this.bar4 = defaults.bar4;
+    	      this.bar40 = defaults.bar40;
+    	      this.bar41 = defaults.bar41;
+    	      this.bar42 = defaults.bar42;
+    	      this.bar43 = defaults.bar43;
+    	      this.bar44 = defaults.bar44;
+    	      this.bar45 = defaults.bar45;
+    	      this.bar46 = defaults.bar46;
+    	      this.bar47 = defaults.bar47;
+    	      this.bar48 = defaults.bar48;
+    	      this.bar49 = defaults.bar49;
+    	      this.bar5 = defaults.bar5;
+    	      this.bar50 = defaults.bar50;
+    	      this.bar51 = defaults.bar51;
+    	      this.bar52 = defaults.bar52;
+    	      this.bar53 = defaults.bar53;
+    	      this.bar54 = defaults.bar54;
+    	      this.bar55 = defaults.bar55;
+    	      this.bar56 = defaults.bar56;
+    	      this.bar57 = defaults.bar57;
+    	      this.bar58 = defaults.bar58;
+    	      this.bar59 = defaults.bar59;
+    	      this.bar6 = defaults.bar6;
+    	      this.bar60 = defaults.bar60;
+    	      this.bar61 = defaults.bar61;
+    	      this.bar62 = defaults.bar62;
+    	      this.bar63 = defaults.bar63;
+    	      this.bar64 = defaults.bar64;
+    	      this.bar65 = defaults.bar65;
+    	      this.bar66 = defaults.bar66;
+    	      this.bar67 = defaults.bar67;
+    	      this.bar68 = defaults.bar68;
+    	      this.bar69 = defaults.bar69;
+    	      this.bar7 = defaults.bar7;
+    	      this.bar70 = defaults.bar70;
+    	      this.bar71 = defaults.bar71;
+    	      this.bar72 = defaults.bar72;
+    	      this.bar73 = defaults.bar73;
+    	      this.bar74 = defaults.bar74;
+    	      this.bar75 = defaults.bar75;
+    	      this.bar76 = defaults.bar76;
+    	      this.bar77 = defaults.bar77;
+    	      this.bar78 = defaults.bar78;
+    	      this.bar79 = defaults.bar79;
+    	      this.bar8 = defaults.bar8;
+    	      this.bar80 = defaults.bar80;
+    	      this.bar81 = defaults.bar81;
+    	      this.bar82 = defaults.bar82;
+    	      this.bar83 = defaults.bar83;
+    	      this.bar84 = defaults.bar84;
+    	      this.bar85 = defaults.bar85;
+    	      this.bar86 = defaults.bar86;
+    	      this.bar87 = defaults.bar87;
+    	      this.bar88 = defaults.bar88;
+    	      this.bar89 = defaults.bar89;
+    	      this.bar9 = defaults.bar9;
+    	      this.bar90 = defaults.bar90;
+    	      this.bar91 = defaults.bar91;
+    	      this.bar92 = defaults.bar92;
+    	      this.bar93 = defaults.bar93;
+    	      this.bar94 = defaults.bar94;
+    	      this.bar95 = defaults.bar95;
+    	      this.bar96 = defaults.bar96;
+    	      this.bar97 = defaults.bar97;
+    	      this.bar98 = defaults.bar98;
+    	      this.bar99 = defaults.bar99;
+        }
+
+        @CustomType.Setter
```

Comment:
Why does it need both @CustomType.Setter and @CustomType.Parameter - curious?

Reply:
Please be mindful of gigantic generated code sizes that remain a large problem for the speed of Pulumi experience. If something can be solved with fewer API elements we should probably consider that as it puts less pressure on the generated code. I don't have the numbers so perhaps this is negligible, but you could theoretically take into account that setters will only have 1 parameter, and record the name of said parameter in `@CustomType.Setter`. You could also skip recording the name if it happens to match a reflectable name. 

That said I'm good with merging this for now, just consider generated code size when making further changes in the future.

Reply:
> You could also skip recording the name if it happens to match a reflectable name.

Unfortunately this is not possible, we change the name of the setter in codegen in some cases.

Yes, you are right, I could reuse the same annotation. I'll make a followup PR with this.

Reply:
the follow up: https://github.com/pulumi/pulumi-java/pull/730